src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,65 @@
+import { cloneElement, FunctionComponent, useRef } from 'react'
+import { IModalProps } from './IModalProps'
+import DefaultCloseIcon from './DefaultCloseIcon'
+import './Modal.css'
+
+const Modal : FunctionComponent<IModalProps> 
+= ({
+    showModal,
+    onClose,
+    CloseIcon = <DefaultCloseIcon />,
+    title,
+    footer,
+    rounded,
+    children
+}) => {
+    const modalRef = useRef<HTMLDivElement>(null)
+
+    const handleOnClose = () => {
+        onClose()
+    }
+
+    const handleOnClick = (e : any) => {
```

Comment:
If you make the backdrop a sibling of the modal you don't need to do all this ref magic :)